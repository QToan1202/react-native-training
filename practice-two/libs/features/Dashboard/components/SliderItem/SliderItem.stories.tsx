import { ComponentMeta, ComponentStory } from '@storybook/react-native'
import { action } from '@storybook/addon-actions'

import { SliderItem } from '../index'

export default {
  title: 'components/Slider Item',
  component: SliderItem,
} as ComponentMeta<typeof SliderItem>

const Template: ComponentStory<typeof SliderItem> = (args) => <SliderItem {...args} />

export const Default = Template.bind({})
Default.args = {
  source: require('@assets/slider/deliver.png'),
  title: 'Ready to deliver to your home',
  btnTitle: 'Start Shopping',
  onPress: action('press'),
}
