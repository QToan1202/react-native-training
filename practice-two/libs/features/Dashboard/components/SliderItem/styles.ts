import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    paddingTop: 50,
    paddingBottom: 32,
    minWidth: 300,
    alignSelf: 'flex-start',
    rowGap: 20,
    borderRadius: 8,
  },
})

export default styles
