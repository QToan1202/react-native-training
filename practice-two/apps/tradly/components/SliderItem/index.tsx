import { memo } from 'react'
import isEqual from 'react-fast-compare'
import {
  ImageBackground,
  ImageBackgroundProps,
  ImageStyle,
  StyleProp,
  ViewStyle,
} from 'react-native'

import { Paragraph, Button } from '@practice-two/shared'
import styles from './styles'

export interface SliderItemProps extends ImageBackgroundProps {
  source: ImageBackgroundProps['source']
  title: string
  btnTitle: string
  style?: StyleProp<ViewStyle>
  imageStyle?: StyleProp<ImageStyle>
  onPress: () => void
}

const SliderItem = ({
  source,
  title,
  btnTitle,
  style,
  imageStyle,
  onPress,
  ...rest
}: SliderItemProps) => {
  return (
    <ImageBackground
      source={source}
      style={[styles.container, style]}
      imageStyle={imageStyle}
      {...rest}
    >
      <Paragraph
        content={title}
        maxWidth={200}
        marginLeft="$space.4"
        lineHeight="$2"
        letterSpacing="$6"
        fontWeight="$3"
        textTransform="uppercase"
        textOverflow="ellipsis"
      />
      <Button
        shrink
        title={btnTitle}
        marginLeft="$space.3"
        paddingVertical={4}
        paddingHorizontal="$space.3"
        variant="tertiary"
        fontWeight="$4"
        lineHeight="$4"
        fontSize={12}
        textTransform="uppercase"
        pressStyle={{ opacity: 0.6 }}
        onPress={onPress}
      />
    </ImageBackground>
  )
}

export default memo(SliderItem, isEqual)
