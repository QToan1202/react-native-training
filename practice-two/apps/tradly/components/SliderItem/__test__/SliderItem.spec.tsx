import { render, fireEvent, screen } from '@testing-library/react-native'

import SliderItem from '../index'

const mockOnPress = jest.fn()

describe('Testing Slider Item component', () => {
  beforeEach(() => {
    render(
      <SliderItem
        source={require('@assets/slider/deliver.png')}
        title="my item"
        btnTitle="buy"
        onPress={mockOnPress}
        testID="slide-item"
      />
    )
  })

  it('Render Slider Item component success', () => {
    const component = render(
      <SliderItem
        source={require('@assets/slider/deliver.png')}
        title="my item"
        btnTitle="buy"
        onPress={mockOnPress}
      />
    ).toJSON()

    expect(component).toMatchSnapshot()
  })

  it('Slider Item must have Title and Button component', () => {
    const titleElement = screen.getByText('my item')
    const buttonElement = screen.getByText('buy')

    expect(titleElement).toBeOnTheScreen()
    expect(buttonElement).toBeOnTheScreen()
  })

  it('Slider button can react to user action', () => {
    const buttonElement = screen.getByText('buy')

    fireEvent.press(buttonElement)
    expect(mockOnPress).toBeCalledTimes(1)
  })
})
