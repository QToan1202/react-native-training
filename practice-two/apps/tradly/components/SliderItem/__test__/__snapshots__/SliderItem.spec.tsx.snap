// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Slider Item component Render Slider Item component success 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 8,
        "minWidth": 300,
        "paddingBottom": 32,
        "paddingTop": 50,
        "rowGap": 20,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/slider/deliver.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontFamily": "Montserrat_400Regular",
        },
        {
          "fontSize": 14,
        },
        {
          "fontFamily": "Montserrat_600SemiBold",
          "marginLeft": 17,
          "maxWidth": 200,
          "textTransform": "uppercase",
        },
      ]
    }
  >
    my item
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "transparent",
        "borderColor": "#fff",
        "borderRadius": 50,
        "borderWidth": 2,
        "marginLeft": 12,
        "opacity": 1,
        "paddingHorizontal": 13,
        "paddingVertical": 4,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 6,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
              "fontFamily": "Montserrat_400Regular",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "color": "#fff",
              },
              {
                "fontFamily": "Montserrat_500Medium",
                "textAlign": "center",
                "textTransform": "capitalize",
              },
              {
                "fontFamily": "Montserrat_700Bold",
                "fontSize": 12,
                "textTransform": "uppercase",
              },
            ],
          ]
        }
      >
        buy
      </Text>
    </View>
  </View>
</View>
`;
